### PR TITLE
Add `size` param to `splitPane` action, `split-pane` subcommand

### LINKED ISSUE #6298: `wt.exe` should support `-%,--percent` for setting a pane's size on the commandline
See also #4632. 

This was in the spec but hasn't been implemented yet. Basically just `wt ; split-pane -% 30 cmd` to open the Terminal with two panes, one that's 70% of the available size, and the other that's got 30%. 

Or `wt ; split-pane -% 30 cmd, split-pane -% 70 wsl` for three panes, one with 70/30 split, and the 30 is then split into 30/70, kinda like:
```
+-------------------+-------+
|                   |       |
|                   |       |
|                   +-------+
|                   |       |
|                   |       |
|                   |       |
|                   |       |
|                   |       |
|                   |       |
+-------------------+-------+

```

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Adds a `size` parameter to `splitPane`. This takes a `float`, and specifies the portion of the parent pane that should be used to create the new one. 

This also adds the param to the `split-pane` subcommand.

### Examples
 
| commandline | result |
| -- | -- |
| `wt ; sp -s .25` | image | 
| `wt ; sp -s .8` | image |
| `wt ; sp -s .8 ; sp -H -s .3` | image |

## PR Checklist
* [x] Closes #6298
* [x] I work here
* [x] Tests added/passed
* [x] Docs PR: MicrosoftDocs/terminal#208

## Detailed Description of the Pull Request / Additional comments

I went with `size`, `--size,-s` rather than `percent`, because the arg is the (0,1) version of the size, not the (0%,100%) version. 

## Validation Steps Performed

Added actions, played with the commandline, ran tests

### CODE REVIEW

File: src/cascadia/TerminalApp/Pane.cpp [hunk 1]
```diff
@@ -1271,6 +1271,7 @@ bool Pane::CanSplit(SplitState splitType)
 // Note:
 // - This method is highly similar to Pane::PreCalculateAutoSplit
 std::optional<bool> Pane::PreCalculateCanSplit(const std::shared_ptr<Pane> target,
+                                               // TODO: const float splitSize
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:1274]: eep?

File: src/cascadia/TerminalApp/Pane.cpp [hunk 2]
```diff
@@ -1465,7 +1472,7 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitState
     _gotFocusRevoker.revoke();
 
     _splitState = actualSplitType;
-    _desiredSplitPosition = Half;
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:1468]: do we get to kill the `Half` constant now?

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 3]
```diff
@@ -268,6 +272,9 @@
   <data name="CmdFocusTabTargetArgDesc" xml:space="preserve">
     <value>Move focus the tab at the given index</value>
   </data>
+  <data name="CmdSplitPaneSizeArgDesc" xml:space="preserve">
+    <value>Specify the size as a percentage of the parent pane. Valid values are between (0,1), exclusive.</value>
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw:276]: `(0,1)` _technically_ implies exclusive, but yeah

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 4]
```diff
@@ -809,7 +811,7 @@ namespace winrt::TerminalApp::implementation
             TermControl newControl{ settings, debugConnection };
             _RegisterTerminalEvents(newControl, *newTabImpl);
             // Split (auto) with the debug tap.
-            newTabImpl->SplitPane(SplitState::Automatic, profileGuid, newControl);
+            newTabImpl->SplitPane(SplitState::Automatic, .5f, profileGuid, newControl);
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:814]: you did `0.5f` everywheere else

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/Pane.cpp:1274]: EEK yep that's a TODO

File: src/cascadia/LocalTests_SettingsModel/CommandTests.cpp [hunk 5]
```diff
@@ -147,7 +148,8 @@ namespace SettingsModelLocalTests
             { "name": "command1", "command": { "action": "splitPane", "split": "vertical" } },
             { "name": "command2", "command": { "action": "splitPane", "split": "horizontal" } },
             { "name": "command4", "command": { "action": "splitPane" } },
-            { "name": "command5", "command": { "action": "splitPane", "split": "auto" } }
+            { "name": "command5", "command": { "action": "splitPane", "split": "auto" } },
```
Comment (carlos-zamora) [inline src/cascadia/LocalTests_SettingsModel/CommandTests.cpp:151]: nit: maybe add a test for "split" _and_ "size" both defined?

**Review by carlos-zamora — APPROVED** [review]

**Pull request merged**